source/extensions/filters/http/cache/cache_filter.h

```diff
@@ -85,6 +85,9 @@ class CacheFilter : public Http::PassThroughFilter,
   // onHeaders for Range Responses, otherwise initialized by encodeCachedResponse.
   std::vector<AdjustedByteRange> remaining_ranges_;
 
+  // Stores the headers that can be used to vary responses.
+  absl::flat_hash_set<std::string> allowed_vary_headers_;
```

Comment:
Would this be better as a set of LowerCaseString?

Reply:
`LowerCaseString` doesn't have a defined hash function, so we can't use it inside a set.

Reply:
I'll close this discussion, as the type of the allowlist is going to change (see _https://github.com/envoyproxy/envoy/pull/12763#discussion_r479483061_).
We can keep the conversation on that thread.